src/System.Device.Gpio/System/Device/Gpio/Board.cs

```diff
@@ -0,0 +1,265 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+
+namespace System.Device.Gpio
+{
+    public class Board
```

Comment:
Please choose a more specific name (you are only talking about Raspi variants). We'll very likely need the name "Board" for #878.

Reply:
The current way to identify the board cannot be used. 
```string[] cpuInfoLines = File.ReadAllLines(CpuInfoPath);
            Regex regex = new Regex(@"Hardware\s*:\s*(.*)");
            foreach (string cpuInfoLine in cpuInfoLines)
            {
                Match match = regex.Match(cpuInfoLine);
                if (match.Success)
                {
                    if (match.Groups.Count > 1)
                    {
                        if (match.Groups[1].Value == RaspberryPiHardware)
                        {
                            return new RaspberryPi3Driver();
                        }
                        // Commenting out as HummingBoard driver is not implemented yet, will be added back after implementation
                        // https://github.com/dotnet/iot/issues/76
                        // if (match.Groups[1].Value == HummingBoardHardware)
                        // {
                        //     return new HummingBoardDriver();
                        // }
                        return UnixDriver.Create();
                    }
                }

```
It's striclty related to one board, or better to say, to a class of board. RP3 and CM3 share the same processor, but the pinout is completely different.
So we need to better identify the board. Now I only have RPx board, but the class can be extended to check whatever board or, much better, it can be used in #878 too.
I really cannot find a better name.

Reply:
#878 will need the name as abstract base class (but yes, the implementation will be used there and the above method probably moved, but in a much more generic way). 

Since this class only identifies (an extensible set of) boards, what about calling it `BoardIdentification` or so? That will later even be helpful to have the board type detection and the actual board _representation_ separated.

Reply:
I agree